This allows building on amd64 and with gcc above 3.x. Adding support for
other architectures is easy too.

	-mi

--- tnlTypes.h	2004-09-28 14:45:25.000000000 -0400
+++ tnlTypes.h	2008-08-17 10:30:26.000000000 -0400
@@ -285,6 +285,6 @@
 //----------------------------------------------------------------------------------
 
-#if defined(_M_IX86) || defined(i386)
-#  define TNL_CPU_STRING "Intel x86"
+#if defined(_M_IX86) || defined(i386) || defined(__amd64__)
+#  define TNL_CPU_STRING "x86"
 #  define TNL_CPU_X86
 #  define TNL_LITTLE_ENDIAN
@@ -294,5 +294,5 @@
 #    if __GNUC__ == 2
 #      define TNL_GCC_2
-#    elif __GNUC__ == 3
+#    elif __GNUC__ >= 3
 #      define TNL_GCC_3
 #    else
